sql-plugin/src/main/scala/com/nvidia/spark/rapids/datetimeRebaseUtils.scala

```diff
@@ -87,11 +89,16 @@ object DateTimeRebaseUtils {
 
     // Check the timezone of the file if the mode is LEGACY.
     if (mode == DateTimeRebaseLegacy) {
-      val fileTimeZone = lookupFileMeta(SPARK_TIMEZONE_METADATA_KEY)
-      if (fileTimeZone != null && fileTimeZone != "UTC") {
+      val fileTimeZoneId = Option(lookupFileMeta(SPARK_TIMEZONE_METADATA_KEY)).map { str =>
+        DateTimeUtils.getZoneId(str)
+      }.getOrElse {
+        // Use the default JVM time zone for backward compatibility
+        TimeZone.getDefault.toZoneId
```

Comment:
the fallback should be to sqlConf.sessionLocalTimeZone

Reply:
Not accordingf to Spark source, see https://github.com/apache/spark/blob/master/sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/RebaseDateTime.scala#L45

Reply:
But when writing parquet, this is used?
```
          SPARK_TIMEZONE_METADATA_KEY -> SQLConf.get.sessionLocalTimeZone)
```

Reply:
It seems that there is some consistency/bug in Spark? Anyway, since Spark is using default `TimeZone.getDefault.toZoneId` when reading file, we should follow it too.

Reply:
being consistent with Spark's inconsistency is fine